Source/Mosa.Demo.x86/HAL/X86IOPort.cs

```diff
@@ -19,54 +19,54 @@ public X86IOPortReadWrite(ushort address)
 		/// Reads a byte from the IO Port
 		/// </summary>
 		/// <returns></returns>
-		public override byte Read8()
+		public override byte Read8(int offset = 0)
```

Comment:
No, let's not allow an offset parameter. It breaks the safety of the class.

Reply:
The problem is that it's required for so much drivers, so it's kind of mandatory to have it.

Reply:
The idea is the resource manager provides all the IO ports... this ensures safety. With offsets, the code can reference outside of what resources are provided.